tests/test_series.py

```diff
@@ -842,7 +872,7 @@ def test_update() -> None:
     s1.update(pd.Series([0, 2, 12]))
     # Series.update() accepting objects that can be coerced to a Series was added in 1.1.0 https://pandas.pydata.org/docs/whatsnew/v1.1.0.html
     s1.update([1, 2, -4, 3])
-    s1.update([1, "b", "c", "d"])
+    s1.update([1, "b", "c", "d"])  # type: ignore[list-item] # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
I'm not sure what to do here - update is inplace, so I think we have to reject this type-unsafe usage.

Reply:
It's valid usage.  Change the `def update` to use `Scalar` instead of `S1`

Reply:
This means that `pd.Series([0, 2, 12]).update([1, "b", "c", "d"])` is of type `Series[int]`.


If we want to catch bugs, I would rather reject it and have people be aware of this type-unsafe usage. But I'm also happy to allow it (most stubs are far more lenient and avoid falsely flagging an error).

Reply:
I see your point.  Then let's then keep the test, but surround it with a `TYPE_CHECKING_INVALID_USAGE`